src/node_alive/autostarter.py

```diff
@@ -0,0 +1,119 @@
+# ROS
+import os
+import roslaunch
+import rospkg
+import rospy
+from std_msgs.msg import String
+
+# TU/e Robotics
+from node_alive.srv import AutoStarterCommand, AutoStarterCommandRequest, AutoStarterCommandResponse
+
+class AutoStarter(object):
+    """
+    Launches a launch file and waits for a command to either shutdown or to restart.
+    It is also possible to start other launch files that are stored on the parameter server, like the demo's.
+    """
+    launch_files_list = []
+
+    def __init__(self):
+        """
+        Loads the launch files that can be started from the parameter server.
+        Creates a service over which the command will be received, a publisher that publishes the current launch file.
+        Set some variables.
+        """
+        self.launch_files_list = rospy.get_param('~launch_files_list')
+        self._command_service = rospy.Service('auto_starter_command', AutoStarterCommand, self._srv_callback)
+        self._status_pub = rospy.Publisher('current_launch_file', String, queue_size=1, latch=True)
+
+        self.launch = None
```

Comment:
The member attribute name "self.launch" is not really clear. You might want to add a few comments as well.

Reply:
Wrote extra comment for self.launch in a1ef39d19ac8416def3517506f5d571fcb9e1645